### DISCUSSION TITLE
MacOS: Visual glitch when closing tab with Stage Manager enabled

### DISCUSSION THREAD

**Discussion by kenvunz** [ghostty-org/ghostty#4036] [Feature Requests, Ideas] [answered]
When Stage Manager (SM) is enabled, closing one tab will lose the entire app focus, rather stay focused on the remaining tab.

In this particular example, I have 2 windows, one Chrome and one Ghostty, closing one tab and the app will immediately losing the focus.

![CleanShot 2024-12-30 at 11 33 51](https://github.com/user-attachments/assets/dbf57fed-80df-488e-9aaf-3f27f39fa0bc)

**Comment by dmartzol** [comment]
Not sure if this is the same issue I'm experiencing because in my case something similar happens when **opening a new tab**, but it is similar enough so I will post here:

In my case, while using a window manager([Amethyst](https://github.com/ianyh/Amethyst)) **opening a new tab** will make Ghostty lose entire app focus. I can no longer make Ghostty the main window and I have to restart Amethyst to go back to normal.

Not 100% sure if this is Amethyst or Ghostty's fault tho.

**Comment by mitchellh** [comment, marked as answer]
This is fixed on tip.

**Comment by FammasMaz** [comment]
This seems to have reappeared on the latest and the tip. currently on tahoe, but i remember the same issue in sequoia

**Reply by 00-kat** [reply]
Please open a [new discussion]!

[new discussion]: https://github.com/ghostty-org/ghostty/discussions/new?category=issue-triage
